py/server/deephaven/table_listener.py

```diff
@@ -298,6 +305,9 @@ def __init__(self, t: Table, listener: Union[Callable[[TableUpdate, bool], None]
         Raises:
             DHError
         """
+        if not t.is_refreshing:
+            raise DHError(message="table must be a refreshing table.")
```

Comment:
This, right here, is an example of why replay is a bad pattern. If you separate initialization from listening, you can use your initialization on static tables, too.